File: FireflowSuite/model/src/org/fireflow/model/io/IFPDLSerializer.java
package org.fireflow.model.io;

import java.io.IOException;
import java.io.OutputStream;
import java.io.Writer;

import org.fireflow.model.WorkflowProcess;

public interface IFPDLSerializer extends FPDLNames {
	public void serialize(WorkflowProcess workflowProcess, OutputStream out)
			throws IOException, FPDLSerializerException;

	public void serialize(WorkflowProcess workflowProcess, Writer out)
			throws IOException, FPDLSerializerException;
}

File: dot-net_version/FireWorkflow.Net/FireWorkflow.Net/Model/Io/IFPDLSerializer.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using System.Xml;
using System.Xml.Serialization;

namespace FireWorkflow.Net.Model.Io
{
    /// <summary>
    /// FPDL序列化器。将WorkflowProcess对象序列化到一个输出流。
    /// </summary>
    public abstract class IFPDLSerializer : FPDLNames
    {

        /// <summary>
        /// 将WorkflowProcess对象序列化到一个输出流。
        /// </summary>
        /// <param name="workflowProcess">工作流定义</param>
        /// <param name="swout">输出流</param>
        public abstract void serialize(WorkflowProcess workflowProcess, Stream swout);
        /*
	public void serialize(WorkflowProcess workflowProcess, Writer out)
			throws IOException, FPDLSerializerException;
         */

    }
}
